cloudflare/resource_cloudflare_load_balancer.go

```diff
@@ -0,0 +1,335 @@
+package cloudflare
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"time"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/pkg/errors"
+)
+
+func resourceCloudFlareLoadBalancer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudFlareLoadBalancerCreate,
+		Read:   resourceCloudFlareLoadBalancerRead,
+		Update: resourceCloudFlareLoadBalancerUpdate,
+		Delete: resourceCloudFlareLoadBalancerDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudFlareLoadBalancerImport,
+		},
+
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"zone_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false, // allow reusing same load balancer for different dns names
+			},
+
+			"fallback_pool_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 32),
+			},
+
+			"default_pool_ids": {
+				Type:     schema.TypeList,
```

Comment:
Is the ordering of pool ids significant such that it affects the behavior of the load balancer? Will these pool id's always be returned from the API in the same order (assuming no external modifications)?

Reply:
yes they should be, this is intended to define the failover order for these pools

Reply:
👍